### ISSUE TITLE
Checks are not failing when tests or linting fail

### ISSUE THREAD

**Issue by aganders3** [vispy/vispy#2596]
I have noticed a couple issues with the tests perhaps not testing what is expected.

First, I think the linting check in CI is broken. I mentioned it in https://github.com/vispy/vispy/issues/2291#issuecomment-1601356010, but I don't think it was resolved.

Second, which I can't (yet) explain, there is a test that should have failed in #2589 but didn't:
https://github.com/vispy/vispy/blob/4f061ba23ddac3131eb0d1d955573dd71c9a3d2f/vispy/visuals/tests/test_volume.py#L46-L48
I noticed this testing locally (where it did fail). Looking at the [checks for that PR](https://github.com/vispy/vispy/actions/runs/9267063805/job/25492687269?pr=2589#step:11:642) I think the test did fail, but the check passed.

**Comment by djhoese** [comment]
I can't say for sure without taking a long look, but i believe linting checks were disabled because they just all failed and no one had the time to fix them all. There are also Image and Volune tests that are skipped in CI because they depend on modern handling of NaN by the GPU and githubs runners don't do that.

**Comment by aganders3** [comment]
>  i believe linting checks were disabled because they just all failed and no one had the time to fix them all.

Okay, I'm not too worried if that's the case but I just missed the discussion. Last I looked into it, it seemed like it broke due to a change in flake8, because it's invoked here in code and not as a CLI app. Therefore the behavior changed in flake8 5.0 because of https://github.com/PyCQA/flake8/pull/1461.

> There are also Image and Volune tests that are skipped in CI because they depend on modern handling of NaN by the GPU and githubs runners don't do that.

I think these are marked XFail? You have more context for sure though.
https://github.com/vispy/vispy/blob/4f061ba23ddac3131eb0d1d955573dd71c9a3d2f/vispy/visuals/tests/test_image.py#L128

Regardless the test I linked here seems to have run and failed, but for whatever reason did not trigger a failure in the check.
